src/main.js

```diff
@@ -14,6 +14,7 @@ const XML_FILE_EXTENSION = 'xml'
 
 module.exports = config => {
   const cliHelper = new CLIHelper(config)
+  cliHelper.validateGitSha()
```

Comment:
**issue(_blocking_):** the `validateGitSha` should be private to cliHelper and called after `_sanitizeConfig`

In term of API it does not need to expose another method
`_sanitizeConfig` redress the value from `to` and `from` then `_validateGitSha` should be called at the end of `validateConfig` (just before the check on error.length to throw the error array

Reply:
Agree, but there is one problem with this approach. If 'to' is incorrect, the user will get the following error:
`"error": "--generate-delta (-d) parameter cannot be used when --to (-t) parameter is not equivalent to HEAD"`

This will mislead the user by hiding the real culprit, which is the wrong 'to' SHA.

So, I don't mind calling _validateGitSha internally from cliHelper, but this should happen before validateConfig or at least at the beginning of it, e.g. right after `this._sanitizeConfig()`

Reply:
It should be between the check if "to" is a string and the the check if two is equal to head then.
But anyway, as we are storing all the error in the array we will see both error (to not a sha pointer and to different than HEAD with delta generate enabled).
So maybe the right way to deal with that is to have a chain of responsibility in this order:

1. check to is not a string
2. check to is sha pointer
3. check to is different than HEAD with delta generate enabled

And also check from is sha pointer.

Then maybe the best approach is to have a private method (or a static one) in this class to check if a sha pointer exist and then use it where we need in the code

Reply:
Thanks for your feedback, @scolladon. I'll make a fix to this PR. 
Also, there is no 'string' type check for 'from'. What's the reason for this?

Reply:
We do not checked the `--from` parameter because it is optional and it is [computed](https://github.com/scolladon/sfdx-git-delta/blob/main/src/utils/repoSetup.js#L46) when not set (or set to empty string) by the user.
That make me think that the check that `--to` should be rewritten to check that it is not null neither empty (`!!config.to`) because it will always be a string when set but it could be empty which can lead to errors
Could you also embed the new way to check `to` ?

Reply:
Sure. Working on it.

Reply:
@scolladon, I'm a bit confused here: you say `--from` is optional, while [your code](https://github.com/scolladon/sfdx-git-delta/blob/76cc459fa5fb946df3607b0478a2062ce5c89fb0/src/commands/sgd/source/delta.ts#L28) clearly states that it's required:
`    from: flags.string({
      char: 'f',
      description: messages.getMessage('fromFlag'),
      required: true,
    }),
`
And you'll see this error if --from is not provided:
`ERROR running sgd:source:delta:  Missing required flag:
 -f, --from FROM  commit sha from where the diff is done [git rev-list --max-parents=0 HEAD]
See more help with --help`

Conversely, --to is optional and defaults to 'HEAD'. So it doesn't make much sense to check `(!this.config.to)`

Reply:
I wasn't able to pass an empty value to -f. Tried single/double quotes and no quotes at all. In all cases the user will get:
`ERROR running sgd:source:delta:  Flag --from expects a value`
So ` required: true` takes care of all those cases, which makes `computeFromRef()` redundant.
With this in mind, why not just make --to required as well?
Then we won't need any additional checks for 'null or empty' and can go straight to sha verification for the both.

Reply:
Same with --to. Although it is not required, once mentioned in CLI it can't be left w/o some value other than an empty string. `--to ''`, `--to ""`, or `--to ` will all throw:
`ERROR running sgd:source:delta:  Flag --to expects a value`

Reply:
Bottom line:
1. the 'null-or-empty' check is provided by the CLI library, so we don't have to do it.
2. since `--from` is a required parameter and it can't be empty, `computeFromRef()` is redundant.
3. since `--to` has a default value='HEAD', we'll never see it empty either, this makes the `!config.to` check redundant.

So we're left with the following checks for the 2 parameters

For `--to`:
1. if not `_isValidGitSha()` - throw error and don't check #2
2. if `-d` is passed and `--to` != HEAD - throw error

For `--from`:
1. if not `_isValidGitSha()` - throw error

Reply:
This is not what I experience on my side:
![Screenshot 2022-02-24 at 07 26 28](https://user-images.githubusercontent.com/522422/155470053-fa85067e-2f5c-4dfb-b1b6-3b1ba310240b.png)

I manage to set `--to` and `--from` to empty value without the CLI throwing.
My environment is:
```sh
$ uname -v ; yarn -v ; node -v ; git --version ; sfdx --version ; sfdx plugins
Darwin Kernel Version 19.6.0: Thu Jan 13 01:26:33 PST 2022; root:xnu-6153.141.51~3/RELEASE_X86_64
yarn 1.22.17
node v16.13.1
git version 2.35.1
sfdx-cli/7.138.1 darwin-x64 node-v16.13.1
sfdx-git-delta 4.12.1
```

What is yours ?

for me we need to keep
> 1. the 'null-or-empty' check is provided by the CLI library, so we don't have to do it.
> 1. since --from is a required parameter and it can't be empty, computeFromRef() is redundant.
> 1. since --to has a default value='HEAD', we'll never see it empty either, this makes the !config.to check redundant.

'null-or-empty' check `--to` and `--from` need to check for blank character (spaces, tabs, carriage return, etc), empty string, null and undefined.

Reply:
I'm on Windows:
`$  uname -v ; yarn -v ; node -v ; git --version ; sfdx --version ; sfdx plugins
2021-10-12 16:29 UTC
1.22.17
v16.13.2
git version 2.34.1.windows.1
sfdx-cli/7.138.1 win32-x64 node-v16.13.2
sfdx-git-delta 4.12.1 (link) C:\Projects\Salesforce\sfdx-git-delta
`

If libraries behave differently in different OS, then certainly we must keep the in-code verification.